wait/member.go

```diff
@@ -29,30 +29,37 @@ func NewMemberAwaitility(a *Awaitility) *MemberAwaitility {
 		}}
 }
 
-// WaitForUserAccount waits until there is a UserAccount available with the given name, expected spec and the set of status conditions
-func (a *MemberAwaitility) WaitForUserAccount(name string, expSpec toolchainv1alpha1.UserAccountSpec, conditions ...toolchainv1alpha1.Condition) error {
-	return wait.Poll(RetryInterval, Timeout, func() (done bool, err error) {
-		ua := &toolchainv1alpha1.UserAccount{}
+// WaitForUserAccount waits until there is a UserAccount available with the given name and the set of status conditions
+func (a *MemberAwaitility) WaitForUserAccount(name string, conditions ...toolchainv1alpha1.Condition) (*toolchainv1alpha1.UserAccount, error) {
+	return a.WaitForUserAccountWithSpec(name, toolchainv1alpha1.UserAccountSpec{}, conditions...)
+}
+
+// WaitForUserAccountWithSpec waits until there is a UserAccount available with the given name, expected spec and the set of status conditions
+func (a *MemberAwaitility) WaitForUserAccountWithSpec(name string, expSpec toolchainv1alpha1.UserAccountSpec, conditions ...toolchainv1alpha1.Condition) (*toolchainv1alpha1.UserAccount, error) {
+	ua := &toolchainv1alpha1.UserAccount{}
+	err := wait.Poll(RetryInterval, Timeout, func() (done bool, err error) {
 		if err := a.Client.Get(context.TODO(), types.NamespacedName{Namespace: a.Ns, Name: name}, ua); err != nil {
 			if errors.IsNotFound(err) {
 				a.T.Logf("waiting for availability of useraccount '%s'", name)
 				return false, nil
 			}
 			return false, err
 		}
-		if reflect.DeepEqual(ua.Spec, expSpec) &&
+		if (reflect.DeepEqual(expSpec, toolchainv1alpha1.UserAccountSpec{}) || reflect.DeepEqual(ua.Spec, expSpec)) &&
```

Comment:
just a proposal, not a requested change:
Wouldn't be better to introduce "wait conditions" similar to what we have for MUR: https://github.com/codeready-toolchain/toolchain-e2e/pull/31/files#diff-dfba07fbdde19dfb6e7d80277c03139eR51?

Reply:
done in https://github.com/codeready-toolchain/toolchain-e2e/commit/8032a7d38df52c7c09154e1d83cb8083ed08ada1